samples/shopping-cart-with-view/src/main/java/shoppingcart/domain/ShoppingCartState.java

```diff
@@ -0,0 +1,57 @@
+package shoppingcart.domain;
+
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.Comparator;
+import java.util.Optional;
+import java.util.function.Predicate;
+
+// tag::domain[]
+public record ShoppingCartState(String cartId, List<LineItem> items, boolean checkedOut) {
```

Comment:
I think mention about this, but I will good to have a naming symmetry with the "basic" shopping cart sample, so that users can easily follow more complex sample.

Reply:
in this particular case, this rename is one of the refactors that is intentionally made to the sample to clarify names and roles.